internal/database/mariadb/service_issue_variant.go

```diff
@@ -0,0 +1,130 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package mariadb
+
+import (
+	"fmt"
+	"github.com/cloudoperators/heureka/internal/entity"
+	"github.com/jmoiron/sqlx"
+	"github.com/sirupsen/logrus"
+)
+
+func (s *SqlDatabase) ensureServiceIssueVariantFilter(f *entity.ServiceIssueVariantFilter) *entity.ServiceIssueVariantFilter {
+	var first = 1000
+	var after int64 = 0
+	if f == nil {
+		return &entity.ServiceIssueVariantFilter{
+			Paginated: entity.Paginated{
+				First: &first,
+				After: &after,
+			},
+			ComponentInstanceId: nil,
+		}
+	}
+
+	if f.After == nil {
+		f.After = &after
+	}
+	if f.First == nil {
+		f.First = &first
+	}
+	return f
+}
+
+func (s *SqlDatabase) getServiceIssueVariantFilterString(filter *entity.ServiceIssueVariantFilter) string {
+	var fl []string
+	fl = append(fl, buildFilterQuery(filter.ComponentInstanceId, "CI.componentinstance_id = ?", OP_OR))
+	fl = append(fl, "IV.issuevariant_deleted_at IS NULL")
+
+	return combineFilterQueries(fl, OP_AND)
+}
+
+func (s *SqlDatabase) buildServiceIssueVariantStatement(baseQuery string, filter *entity.ServiceIssueVariantFilter, withCursor bool, l *logrus.Entry) (*sqlx.Stmt, []interface{}, error) {
+	var query string
+	filter = s.ensureServiceIssueVariantFilter(filter)
+	l.WithFields(logrus.Fields{"filter": filter})
+
+	filterStr := s.getServiceIssueVariantFilterString(filter)
+	cursor := getCursor(filter.Paginated, filterStr, "IV.issuevariant_id > ?")
+
+	whereClause := ""
+	if filterStr != "" || withCursor {
+		whereClause = fmt.Sprintf("WHERE %s", filterStr)
+	}
+
+	// construct final query
+	if withCursor {
+		query = fmt.Sprintf(baseQuery, whereClause, cursor.Statement)
+	} else {
+		query = fmt.Sprintf(baseQuery, whereClause)
+	}
+
+	//construct prepared statement and if where clause does exist add parameters
+	var stmt *sqlx.Stmt
+	var err error
+
+	stmt, err = s.db.Preparex(query)
+	if err != nil {
+		msg := ERROR_MSG_PREPARED_STMT
+		l.WithFields(
+			logrus.Fields{
+				"error": err,
+				"query": query,
+				"stmt":  stmt,
+			}).Error(msg)
+		return nil, nil, fmt.Errorf("%s", msg)
+	}
+
+	//adding parameters
+	var filterParameters []interface{}
+	filterParameters = buildQueryParameters(filterParameters, filter.ComponentInstanceId)
+
+	if withCursor {
+		filterParameters = append(filterParameters, cursor.Value)
+		filterParameters = append(filterParameters, cursor.Limit)
+	}
+
+	return stmt, filterParameters, nil
+}
+
+func (s *SqlDatabase) GetServiceIssueVariants(filter *entity.ServiceIssueVariantFilter) ([]entity.ServiceIssueVariant, error) {
+	l := logrus.WithFields(logrus.Fields{
+		"event": "database.GetIssueVariants",
+	})
+
+	baseQuery := `
+		SELECT IRS.issuerepositoryservice_priority, IV.*  FROM  ComponentInstance CI 
```

Comment:
does this query assume that the issue is attached to the component version and not to the issue match?

Reply:
Yes this query is especially for the use case of looking up (not yet existing) issue matches